src/js/components/Booking.js

```diff
@@ -113,13 +114,21 @@ class Booking{
     let allAvailable = false;
 
     if(
-      typeof thisBooking.booked[thisBooking.date] == 'undefined'
+      typeof thisBooking.booked[thisBooking.date] === 'undefined'
```

Comment:
Po pierwsze chcesz sprawdzić czy dana **wartość** istnieje w obiekcie czy jest `undefined` dlatego nie ma sensu odwoływać się do `typeof`. 
Druga rzecz to, że wartość `undefined` jest tzw. "falsy value" (https://developer.mozilla.org/en-US/docs/Glossary/Falsy) a więc najlepiej będzie to zapisać 
if (!thisBooking.booked[thisBooking.date]) {...} - co będzie znaczyć - jeśli wartość jest falsy (np. undefined/null/false itp) to wejdź w if'a.